# Simulation settings for using moveit_sim_controllers
moveit_sim_hw_interface:
  joint_model_group: hand
  joint_model_group_pose: open_gripper
# Settings for ros_control_boilerplate control loop
generic_hw_control_loop:
  loop_hz: 300
  cycle_time_error_threshold: 0.01
# Settings for ros_control hardware interface
hardware_interface:
  joints:
    #- front_left_wheel_joint
    #- front_right_wheel_joint
    - joint_1
    - joint_2
    - joint_3
    - eff_joint
    - left_finger_joint
    - right_finger_joint
  sim_control_mode: 1  # 0: position, 1: velocity
# Publish all joint states
# Creates the /joint_states topic necessary in ROS


joint_state_controller:
  type: joint_state_controller/JointStateController
  publish_rate: 50
hand_controller:
  type: position_controllers/JointTrajectoryController
  joints:
    - left_finger_joint
    - right_finger_joint
  gains:
    left_finger_joint: { p: 100, d: 1, i: 1, i_clamp: 1}
    right_finger_joint: { p: 100, d: 1, i: 1, i_clamp: 1}
arm_controller:
  type: position_controllers/JointTrajectoryController
  joints:
    - joint_1
    - joint_2
    - joint_3
    - eff_joint
  gains:
    joint_1: { p: 100, d: 1, i: 1, i_clamp: 1}
    joint_2: { p: 100, d: 1, i: 1, i_clamp: 1}
    joint_3: { p: 100, d: 1, i: 1, i_clamp: 1}
    eff_joint: { p: 100, d: 1, i: 1, i_clamp: 1}
  

controller_list:
  - name: hand_controller
    action_ns: follow_joint_trajectory
    type: FollowJointTrajectory
    default: true
    joints:
      - left_finger_joint
      - right_finger_joint
  - name: arm_controller
    action_ns: follow_joint_trajectory
    type: FollowJointTrajectory
    default: true
    joints:
      - joint_1
      - joint_2
      - joint_3
      - eff_joint
